Automattic/wp-calypso
# Developer UX: create HOC withScrollTopOnClick for SMP navigation

- Related https://github.com/Automattic/dotcom-forge/issues/1635

## Proposed Changes

* Create HOC `withScrollTopOnClick`
* Scroll to the Top when SMP action menu items are clicked (`MenuItemLink`).

## Testing Instructions



IT FAILS THE TEST.

* Go to SMP `/sites`
* Scroll down the page
* Click on the actions menu on a business site
* Click on `Performance settings` or `Hosting configuration`
* Observe you've navigated to the correct page but the scroll is NOT at the top.


## Screencast

https://user-images.githubusercontent.com/779993/221894001-6651a4ee-1679-4e5d-a3f3-b71d6f4dd202.mp4




## Pre-merge Checklist
